src/azure-cli/azure/cli/command_modules/rdbms/_params.py

```diff
@@ -153,17 +170,228 @@ def _complex_params(command_group):
+            c.argument('vnet_address_prefix', options_list=['--vnet-address-prefix'], help='The virtual network address prefix.')
+            c.argument('subnet_address_prefix', options_list=['--subnet-address-prefix'], help='The subnet address prefix.')
+            c.argument('subnet_arm_resource_id', options_list=['--subnet'],
+                       help='Name or ID of the subnet that allows access to an Azure Flexible Server. ')
+            c.argument('server_name', options_list=['--name', '-n'], arg_type=server_name_setter_arg_type)
+            c.argument('location', arg_type=get_location_type(self.cli_ctx))
+            c.argument('administrator_login', default=generate_username(), options_list=['--admin-user, -u'], arg_group='Authentication', arg_type=administrator_login_setter_arg_type,
+                       help='Administrator username for the server. Once set, it cannot be changed. ')
+            c.argument('administrator_login_password', options_list=['--admin-password', '-p'],
+                       help='The password of the administrator. Minimum 8 characters and maximum 128 characters. Password must contain characters from three of the following categories: English uppercase letters, English lowercase letters, numbers, and non-alphanumeric characters.',
+                       arg_group='Authentication')
+            c.argument('backup_retention', default=7, type=int, options_list=['--backup-retention'],
+                       help='The number of days a backup is retained. Range of 7 to 35 days. Default is 7 days.',
+                       validator=retention_validator)
+            c.argument('tags', tags_type)
+            c.argument('public_access', options_list=['--public-access'],
+                       help='Determines the public access. Enter single or range of IP addresses to be included in the allowed list of IPs. IP address ranges must be dash-separated and not contain any spaces. Specifying 0.0.0.0 allows public access from any resources deployed within Azure to access your server. Specifying no IP address sets the server in public access mode but does not create a firewall rule. ',
+                       validator=public_access_validator)
+            c.argument('high_availability', default="Disabled", options_list=['--high-availability'], help='Enable or disable high availability feature.  Default value is Disabled.')
+            c.argument('assign_identity', options_list=['--assign-identity'],
+                       help='Generate and assign an Azure Active Directory Identity for this server for use with key management services like Azure KeyVault. No need to enter extra argument.')
+            c.ignore('database_name')
+
+        with self.argument_context('{} flexible-server delete'.format(command_group)) as c:
+            c.argument('server_name', id_part='name', options_list=['--name', '-n'], arg_type=server_name_getter_arg_type)
```

Comment:
with self.argument_context('{} flexible-server list'.format(command_group)) as c:
            c.argument('server_name', id_part=None, arg_type=server_name_getter_arg_type)
 
All "list" commands should do like above to avoid the following error "`mysql flexible-server db list` - List commands should not expose --ids argumen"

Reply:
Thanks, fixing this and merging.